packages/xo-server/src/xo-mixins/metadata-backups.js

```diff
@@ -349,17 +349,12 @@ export default class metadataBackup {
           }
         )
 
-        let outputStream
         try {
           await waitAll([
             (async () => {
-              outputStream = await handler.createOutputStream(fileName)
-
-              // 'readable-stream/pipeline' not call the callback when an error throws
-              // from the readable stream
-              stream.pipe(outputStream)
               return timeout.call(
-                fromEvent(stream, 'end').catch(error => {
+                handler.outputStream(fileName, stream).catch(error => {
+                  stream.destroy()
```

Comment:
@julien-f what do you think of that?

Reply:
I don't think it will properly work, the file will not be cleaned up by `outputStream`.

I think the proper way would be to add a `{ cancelToken }` param to `outputStream` to properly abort the operation.

See https://github.com/JsCommunity/promise-toolbox#cancelation